### PR TITLE
Feature/add validation and specs

### PR DESCRIPTION
**Pull request by jaferIdrees**
### On this branch I completed the below tasks:

- Add the following validations for the User model:

- [ ] Name must not be blank.
- [ ] PostsCounter must be an integer greater than or equal to zero.

- Add the following validations for the Post model:

- [ ] Title must not be blank.
- [ ] Title must not exceed 250 characters.
- [ ] CommentsCounter must be an integer greater than or equal to zero.
- [ ] LikesCounter must be an integer greater than or equal to zero.

- [ ] Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @jaferIdrees 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### Highlights

✔️ Meaningful Commit Messages
✔️ Descriptive PR
✔️ Proper Gitflow

## Required Changes ♻️

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jaferIdrees** [comment]
Hi @Whoistolu ,

I can't see your comments or required changes!

File: spec/model/post_spec.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    it 'first_user.recent_posts should return three records' do
+      expect(first_post.recent_comments.size).to eq(5)
+    end
+
+    it 'first_post.comments_counter should equal 6' do
+      expect(first_post.comments_counter).to eq(6)
+    end
+  end
+end
```
Comment (Whoistolu) [inline spec/model/post_spec.rb:46]: - [ ] [REQUIRED] Hello @jaferIdrees , Apologies for the inconvenience. I had initially given an inline reference here earlier but I really do not know why it didn't display.
I had stated that you did a great implementation writing the unit tests for the `Post` and the `User` model however, the project requirement states that you should write unit tests for the `Comment` and the `Like` model also.

**Review by Whoistolu — CHANGES_REQUESTED** [review]
Hi @jaferIdrees 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️

### Highlight
✔️ Descriptive PR
✔️ Meaningful Commit Message

- [ ]  _Check the inline comments under the review for sections to improve._

### Optional suggestions
Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

### N/A

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by bimbolabuari — COMMENTED** [review]
Hello @jaferIdrees  👋,

_This is an additional review as a result of the second opinion request._

- Kudos to you @Whoistolu for rectifying the issue 👍 

- @jaferIdrees, kindly make the requested changes and ask for another review to get approved 👍 

Keep rocking 🚀!

**Review by alexander16108 — CHANGES_REQUESTED** [review]
Hi @jaferIdrees , 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

You have submitted a project, that has linter errors take step back and try again

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

**Review by tufoinnkuo10 — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi @jaferIdrees 

Your project is complete! Great job for making the requested changes🥇. There is nothing else to say other than... it's time to merge it 🍾🚢 :shipit: . Congratulations!  🎉💯🌟 

## Good Points 👍
- Descriptive Pull Request
- Unit Test ok
- No linter errors

<br>
Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have 2 more limited reviews per this project. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request closed without merging**